source/material_model/utilities.cc

```diff
@@ -538,29 +552,76 @@ namespace aspect
+          // First two columns should be P(bar) and T(K).
+          // Here we find the order.
+          std::string column_name;
+          in >> column_name;
+
+          std::string first_natural_variable;
+          if (column_name == "P(bar)")
             {
-              std::string label;
-              in >> label;
-              if (label == "rho,kg/m3")
+              first_natural_variable = column_name;
+              in >> column_name;
+              AssertThrow(column_name == "T(K)", ExcMessage("The second column name in PerpleX lookup file " + filename + " should be T(K)."))
+            }
+          else if (column_name == "T(K)")
+            {
+              first_natural_variable = column_name;
+              in >> column_name;
+              AssertThrow(column_name == "P(bar)", ExcMessage("The second column name in PerpleX lookup file " + filename + " should be T(K)."))
+            }
+          else
+            {
+              AssertThrow(false, ExcMessage("The first column name in PerpleX lookup file " + filename + " should be P(bar) or T(K)."))
+            }
+
+          for (unsigned int n=2; n<n_columns; n++)
+            {
+              in >> column_name;
+              if (column_name == "rho,kg/m3")
                 prp_indices[0] = n;
-              else if (label == "alpha,1/K")
+              else if (column_name == "alpha,1/K")
                 prp_indices[1] = n;
-              else if (label == "cp,J/K/kg")
+              else if (column_name == "cp,J/K/kg")
                 prp_indices[2] = n;
-              else if (label == "vp,km/s")
+              else if (column_name == "vp,km/s")
                 prp_indices[3] = n;
-              else if (label == "vs,km/s")
+              else if (column_name == "vs,km/s")
                 prp_indices[4] = n;
-              else if (label == "h,J/kg")
+              else if (column_name == "h,J/kg")
                 prp_indices[5] = n;
+              else if (column_name.length() > 3)
+                {
+                  if (column_name.substr(0,13).compare("vol_fraction_") == 0)
+                    {
+                      if (std::find(phase_column_names.begin(),
+                                    phase_column_names.end(),
+                                    column_name) != phase_column_names.end())
+                        {
+                          AssertThrow(false,
+                                      ExcMessage("The PerpleX lookup file " + filename + " must have unique column names. "
+                                                 "Sometimes, the same phase is stable with >1 composition at the same "
+                                                 "pressure and temperature, so you may see several columns with the same name. "
+                                                 "Either combine columns with the same name, or change the names."));
+                        }
+                      else
```

Comment:
You  don't have to put this into an else statement. If the assert triggers, we never get here anyway.

Reply:
:D